### PR TITLE
Fix massa-client read only call

### PR DESCRIPTION
**Pull request by modship**
* [ ] document all added functions
* [ ] try in sandbox /simulation/labnet
  * [ ] if part of node-launch, checked using the `resync_check` flag
* [x] unit tests on the added/changed features
  * [x] make tests compile
  * [x] make tests pass 
* [ ] add logs allowing easy debugging in case the changes caused problems
* [ ] if the API has changed, update the API specification

### CODE REVIEW

**Comment by modship** [comment]
At the moment, this not working correctly; there is a serialization error.
`Error: check if your node is running: ErrorObject { code: ServerError(-32080), message: "Parse error: invalid type: sequence, expected a map at line 1 column 558", data: None }`

**Comment by modship** [comment]
problem with : https://github.com/massalabs/massa/blob/104d53b5a2b84989ba08f1158e0fbfcadb958af2/massa-ledger-exports/src/ledger_changes.rs#L35

File: massa-ledger-exports/src/ledger_changes.rs [hunk 1]
```diff
@@ -20,38 +20,24 @@ use nom::error::{context, ContextError, ParseError};
...
 #[derive(Default, Debug, Clone, PartialEq, Eq, Deserialize, Serialize)]
 pub struct LedgerEntryUpdate {
     /// change the balance
     pub balance: SetOrKeep<Amount>,
     /// change the executable bytecode
     pub bytecode: SetOrKeep<Bytecode>,
     /// change datastore entries
-    #[serde(serialize_with = "as_array")]
+    #[serde_as(as = "Vec<(_, _)>")]
     pub datastore: BTreeMap<Vec<u8>, SetOrDelete<Vec<u8>>>,
```
Comment (Leo-Besancon) [inline massa-ledger-exports/src/ledger_changes.rs:38]: Where do we use serde serialization for this struct? Would it not be better to implement our own (de)serializer?

Comment (modship) [reply massa-ledger-exports/src/ledger_changes.rs:38]: he has been added here : https://github.com/massalabs/massa/pull/4332

Fix also : https://github.com/massalabs/massa/issues/4428

Comment (Leo-Besancon) [reply massa-ledger-exports/src/ledger_changes.rs:38]: Ok thanks!

**Review by aoudiamoncef — DISMISSED** [review]
LGTM

**Comment by modship** [comment]
bors merge

**Review by Leo-Besancon — APPROVED** [review]

**Comment by modship** [comment]
bors merge

**Pull request merged**